semantic_model_generator/data_processing/cte_utils.py

```diff
@@ -125,6 +161,81 @@ def _generate_cte_for(
         return cte
 
 
+def get_all_physical_column_references(
+    column: semantic_model_pb2.Column,
+) -> List[str]:
+    """Returns a set of column names referenced in the column expression.
+
+    For example, the following column expressions yield the following return values:
+    foo -> [foo]
+    foo+bar -> [foo, bar]
+    sum(foo) -> [foo]
+    """
+    try:
+        parsed = sqlglot.parse_one(column.expr, dialect="snowflake")
+        col_names = set()
+        for col in parsed.find_all(sqlglot.expressions.Column):
+            # TODO(renee): update to use _standardize_sf_identifier to handle quoted columns.
+            col_name = col.name.lower()
+            if col.this.quoted:
+                col_name = col.name
+            col_names.add(col_name)
+        return sorted(list(col_names))
+    except Exception as ex:
+        raise ValueError(f"Failed to parse sql expression: {column.expr}. Error: {ex}")
+
+
+def get_all_column_references_in_table(
+    table: semantic_model_pb2.Table,
+) -> Dict[str, Optional[semantic_model_pb2.Column]]:
+    """Returns all physical table columns referenced in the table.
+
+    Maps the raw table column name to the semantic context Column that directly references it.
+    If the raw table column name has no exact semantic context Column, it'll be mapped to None.
+    """
+    all_col_references: Dict[str, Optional[semantic_model_pb2.Column]] = {}
```

Comment:
Please use more descriptive names otherwise it's difficult to distinguish between logical and physical columns.

Reply:
E.g.  s/all_col_references/physical_to_logical_cols